CBLClient/Apps/CBLTestServer-Android/app/src/main/java/com/couchbase/CouchbaseLiteServ/server/RequestHandler/PredictiveQueriesRequestHandler.java

```diff
@@ -0,0 +1,170 @@
+import com.couchbase.lite.PredictionFunction;*/
+import com.couchbase.lite.*;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+
+public class PredictiveQueriesRequestHandler {
+
+
+  public EchoModel registerModel(Args args){
+    String modelName = args.get("model_name");
+    EchoModel echoModel = new EchoModel(modelName);
+    Database.prediction.registerModel(modelName, echoModel);
+    return echoModel;
+  }
+
+  public void unRegisterModel(Args args){
+    String modelName = args.get("model_name");
+    Database.prediction.unregisterModel(modelName);
+  }
+
+  public List<Object> getPredictionQueryResult(Args args) throws CouchbaseLiteException {
+    EchoModel echoModel = args.get("model");
+    Database database = args.get("database");
+    Map<String, Object> dict = args.get("dictionary");
+    Expression input = Expression.value(dict);
+    PredictionFunction prediction = Function.prediction(echoModel.getName(), input);
+    //String select_property1 = args.get("select_property1");
+    //String select_property2 = args.get("select_property2");
+
+    Query query = QueryBuilder
+        .select(SelectResult.expression(prediction))
+        .from(DataSource.database(database));
+
+    List<Object> resultArray = new ArrayList<>();
+    ResultSet rows = query.execute();
+    for (Result row : rows){
+      resultArray.add(row.toMap());
+    }
+    return resultArray;
+  }
+
+  public String nonDictionary(Args args) throws CouchbaseLiteException {
+    EchoModel echoModel = args.get("model");
+    Database database = args.get("database");
+    String dict = args.get("dictionary");
+    Expression input = Expression.value(dict);
+    PredictionFunction prediction = Function.prediction(echoModel.getName(), input);
+
+    Query query = QueryBuilder
+        .select(SelectResult.expression(prediction))
+        .from(DataSource.database(database));
+
+    List<Object> resultArray = new ArrayList<>();
+    try {
+      query.execute();
+    }
+    catch(Exception e){
+      return e.getLocalizedMessage();
+    }
+    return "success";
+  }
+
+  public List<Object> euclideanDistance(Args args) throws CouchbaseLiteException {
+    Database database = args.get("database");
+
+    String key1 = args.get("key1");
+    String key2 = args.get("key2");
+    Expression distance = Function.euclideanDistance(Expression.property(key1), Expression.property(key2));
+
+    Query query = QueryBuilder
+        .select(SelectResult.expression(distance))
+        .from(DataSource.database(database));
+
+    List<Object> resultArray = new ArrayList<>();
+    ResultSet rows = query.execute();
+    for (Result row : rows){
+      resultArray.add(row.toMap());
+    }
+    return resultArray;
+  }
+
+  public int getNumberOfCalls(Args args) {
+    EchoModel echoModel = args.get("model");
+    return echoModel.numberOfCalls;
+  }
+
+  public List<Object> squaredEuclideanDistance(Args args) throws CouchbaseLiteException {
```

Comment:
Would be better if prefix it with get or set.

Reply:
added